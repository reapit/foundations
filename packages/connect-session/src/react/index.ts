import React from 'react'
import { ReapitConnectHook, ReapitConnectSession } from '../types'
import { ReapitConnectBrowserSession } from '../browser'

export const useReapitConnect = (reapitConnectBrowserSession: ReapitConnectBrowserSession): ReapitConnectHook => {
  const [connectSession, setConnectSession] = React.useState<ReapitConnectSession | null>(null)

  React.useEffect(() => {
    const connectGetSession = async () => {
      const session = await reapitConnectBrowserSession.connectSession()
      if (session) {
        setConnectSession(session)
      }
    }

    connectGetSession()
  }, [])

  const connectAuthorizeRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectAuthorizeRedirect()
  }, [])

  const connectLoginRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLoginRedirect()
  }, [])

  const connectLogoutRedirect = React.useCallback(() => {
    reapitConnectBrowserSession.connectLogoutRedirect()
  }, [])

  const connectIsDesktop = reapitConnectBrowserSession.connectIsDesktop

  const connectHasSession = reapitConnectBrowserSession.connectHasSession

  return {
    connectSession,
    connectAuthorizeRedirect,
    connectLoginRedirect,
    connectLogoutRedirect,
    connectIsDesktop,
    connectHasSession,
  }
}
