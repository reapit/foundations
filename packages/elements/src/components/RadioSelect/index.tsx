import React from 'react'
import { cx } from 'linaria'
import { Field, FieldProps, FormikProps } from 'formik'
import { checkError } from '../../utils/form'
import { Grid, GridItem } from '../Layout'

export type RadioSelectOption = {
  label: string | number
  value: string | number
  additionalField?: (form: FormikProps<any>) => any | React.ReactElement
}

export type RadioSelectProps = {
  name: string
  labelText?: string
  subText?: string
  id: string
  dataTest?: string
  options: RadioSelectOption[]
  setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void
  state: any
  disabled?: boolean
  className?: string
  isHorizontal?: boolean
}

export const renderAdditionalField = (additionalField, form: FormikProps<any>) => {
  if (!additionalField) return null
  if (typeof additionalField === 'function') return additionalField(form)
  return additionalField
}

export const RadioSelect: React.FC<RadioSelectProps> = ({
  name,
  labelText,
  subText,
  id,
  dataTest,
  options,
  setFieldValue,
  state,
  disabled = false,
  isHorizontal = false,
  className = '',
}) => {
  const radioGridClassName = cx(!isHorizontal && 'is-gapless')
  const radioGridItemClassName = cx(isHorizontal ? 'is-narrow' : 'is-full')

  return (
    <Field type="radio" name={name}>
      {({ meta, form }: FieldProps<string>) => {
        const hasError = checkError(meta)
        return (
          <div className={`field pb-2 ${className}`}>
            <div className="control">
              <label className="label" htmlFor={id}>
                {labelText}
              </label>
              {subText && <label className="subtext mb-2">{subText}</label>}
              <Grid className={radioGridClassName} isMultiLine>
                {options.map(({ label, value, additionalField }: RadioSelectOption, index: number) => {
                  return (
                    <GridItem key={index} className={radioGridItemClassName}>
                      <div data-test={dataTest} className="radio-wrap">
                        <input
                          id={`${name}${label}`}
                          className={`checkbox ${hasError ? 'checkbox-error' : ''}`}
                          type="radio"
                          key={value}
                          checked={state === value}
                          name={name}
                          value={value}
                          disabled={disabled}
                          onChange={() => setFieldValue(name, value)}
                        />
                        <label htmlFor={`${name}${label}`}>{label}</label>
                      </div>
                      {renderAdditionalField(additionalField, form)}
                    </GridItem>
                  )
                })}
              </Grid>
            </div>
            {hasError && (
              <div className="has-text-danger" data-test="input-error">
                {meta.error}
              </div>
            )}
          </div>
        )
      }}
    </Field>
  )
}

export default RadioSelect
