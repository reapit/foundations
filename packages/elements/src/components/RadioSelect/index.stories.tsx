import React from 'react'

import { storiesOf } from '@storybook/react'
import { RadioSelect } from '.'
import { Formik, Form } from 'formik'
import { action } from '@storybook/addon-actions'
import { Button } from '../Button'
import { Section } from '@/components/Layout'

const mockProps = {
  name: 'mockName',
  labelText: 'mockLabelText',
  id: 'mockId',
  dataTest: 'mockDatatest',
  options: [
    { label: 'label', value: 'value' },
    { label: 'label1', value: 'value1' },
  ],
}

storiesOf('RadioSelect', module)
  .add('Primary', () => {
    return (
      <Section hasPadding={true} style={{ background: '#f5f7f9' }}>
        <Formik
          initialValues={{ [mockProps.name]: '' }}
          onSubmit={values => {
            action('Form Values' + values)
            console.log(values)
          }}
        >
          {({ setFieldValue, values }) => (
            <Form>
              <div className="column is-half-desktop">
                <RadioSelect setFieldValue={setFieldValue} state={values[mockProps.name]} {...mockProps} />
              </div>
              <Button variant={'primary'} type={'submit'}>
                Submit
              </Button>
            </Form>
          )}
        </Formik>
      </Section>
    )
  })
  .add('Horizontal layout', () => {
    return (
      <Section hasPadding={true} style={{ background: '#f5f7f9' }}>
        <Formik
          initialValues={{ [mockProps.name]: '' }}
          onSubmit={values => {
            action('Form Values' + values)
            console.log(values)
          }}
        >
          {({ setFieldValue, values }) => (
            <Form>
              <div className="column is-half-desktop">
                <RadioSelect isHorizontal setFieldValue={setFieldValue} state={values[mockProps.name]} {...mockProps} />
              </div>
              <Button variant={'primary'} type={'submit'}>
                Submit
              </Button>
            </Form>
          )}
        </Formik>
      </Section>
    )
  })
  .add('Disabled', () => (
    <Section hasPadding={true} style={{ background: '#f5f7f9' }}>
      <Formik
        initialValues={{ [mockProps.name]: '' }}
        onSubmit={values => {
          action('Form Values' + values)
          console.log(values)
        }}
      >
        {({ setFieldValue }) => (
          <Form>
            <RadioSelect disabled setFieldValue={setFieldValue} state={'value'} {...mockProps} />
          </Form>
        )}
      </Formik>
    </Section>
  ))
