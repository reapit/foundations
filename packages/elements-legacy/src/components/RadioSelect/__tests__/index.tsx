import React from 'react'
import { render } from '@testing-library/react'
import RadioSelect, { renderAdditionalField } from '../index'
import { Formik, FormikProps } from 'formik'

describe('RadioSelect', () => {
  it('should match snapshot', () => {
    const mockProps = {
      name: 'mockName',
      labelText: 'mockLabelText',
      id: 'mockId',
      dataTest: 'mockDatatest',
      options: [
        { label: 'label', value: 'value' },
        { label: 'label1', value: 'value1' },
      ],
    }
    const wrapper = render(
      <Formik
        initialValues={{}}
        onSubmit={jest.fn()}
        render={() => <RadioSelect setFieldValue={jest.fn()} state={'value'} {...mockProps} />}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
})

describe('renderAdditionalField', () => {
  it('should run correctly', () => {
    const additionalField = jest.fn()
    const form = {} as FormikProps<any>
    renderAdditionalField(additionalField, form)
    expect(additionalField).toBeCalledWith(form)
  })
})
