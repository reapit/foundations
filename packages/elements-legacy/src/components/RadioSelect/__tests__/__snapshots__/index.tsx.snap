// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioSelect should match snapshot 1`] = `
<Field
  name="mockName"
  type="radio"
>
  <Component />
</Field>
`;
